### ISSUE TITLE
Error in dynamic import in web component 

### ISSUE THREAD

**Issue by dman777** [mholt/PapaParse#1008]
I am trying a dynamic import in chrome. I have

```js
      const { default: Papa } = await import('papaparse');
      const csv = Papa.parse(data);
```

but I get the error `Uncaught (in promise) TypeError: Cannot set properties of undefined (setting 'Papa')` which traces to the module code

```js
(function(root, factory)
{
	/* globals define */
	if (typeof define === 'function' && define.amd)
	{
		// AMD. Register as an anonymous module.
		define([], factory);
	}
	else if (typeof module === 'object' && typeof exports !== 'undefined')
	{
		// Node. Does not work with strict CommonJS, but
		// only CommonJS-like environments that support module.exports,
		// like Node.
		module.exports = factory();
	}
	else
	{
		// Browser globals (root is window)
		root.Papa = factory();
	}
	// in strict mode we cannot access arguments.callee, so we need a named reference to
	// stringify the factory method for the blob worker
	// eslint-disable-next-line func-name
}(this, function moduleFactory()
```

This is being used in the lit element library, which is a web component library using shadow dom. Am I doing anything wrong in the above to cause this error?

**Comment by dman777** [comment]
This happens also with a normal import in the browser. This is being done with no typescript or transpiling.... just native ES in browser

**Comment by troptropcontent** [comment]
Hello, 

I am pretty new with javascript

I am working with a rails 7 app, and trying to make this work, 

In my script i have : 

import Papa from "https://unpkg.com/papaparse@latest/papaparse.min.js";

But in my browser console i have the following error : 

`SyntaxError: The requested module 'papaparse' does not provide an export named 'default'`

Is this the same issue ? 🤔 

Any idea of how to make this work ? 🤔

**Comment by j0sh** [comment]
Getting the same - I wonder if it is because of missing esm support? Eg https://github.com/mholt/PapaParse/issues/939 / https://github.com/mholt/PapaParse/pull/875 (note the workarounds suggested in the #939 don't seem to help with dynamic imports as far as I can tell)

**Comment by j0sh** [comment]
Here is what worked for me; the relevant parts were taken from https://github.com/mholt/PapaParse/pull/978 (via nix-shell which I use to set up my development environment)

```sh
    if ! [ -f papaparse.js ]; then
      curl -Lo papaparse-v5.4.0.zip "https://github.com/mholt/PapaParse/archive/refs/tags/5.4.0.zip"
      unzip papaparse-v5.4.0.zip
      pushd PapaParse-5.4.0
      echo "export default globalThis.Papa;" >> papaparse.js
      sed -i "s/function(root,/function(root=window,/" papaparse.js
      esbuild --minify papaparse.js > ../papaparse.js
      popd
    fi
```

Now this is working in my project, a Chrome extension:

```javascript
async function loadPapaParse(){
  const src = chrome.runtime.getURL('papaparse.js');
  const { default:Papa } = await import(src);
  return Papa;
}
```

**Comment by TuringTux** [comment]
@j0sh Thanks for the helpful workaround! For future reference, I needed to install [`esbuild`](https://esbuild.github.io/getting-started/#install-esbuild) using:

```bash
npm install --save-exact --save-dev esbuild
```

This assumes you already have Node.js pre-installed. I also needed to prefix `npx` before the `esbuild` call, i.e. like this:

```bash
...
npx esbuild --minify papaparse.js > ../papaparse.js
...
```

**Comment by markg85** [comment]
I was having this issue too.
The comment from @TuringTux didn't work for me **in a browser context.** I had to use this command:
```bash
npx esbuild --format=esm --minify papaparse.js > papaparse.min.mjs
```

And then import as:
```js
import Papa from "/location/of/papaparse.min.mjs"
```

It does give a parse warning at the `npx esbuild` time but it works so.. that's good enough for me :)

**Comment by mitra42** [comment]
This is a lot of mess to get ESM support in a module that seems to have decided not to support it - is there another CSV parser with ESP support ? 

csv-parser doesn't have ESM support; and I can't figure out how to get csv-parse to look at the results of a fetch.

**Comment by mitra42** [comment]
Yes - I ended up using that first one (note you write `csv-parser` but the link is to `csv-parse` which is the one I used.
